File: ECE39500/exams/bak_midterm-2/PartA/ErrorState.java
public class ErrorState extends CompilerState {

   Compiler compiler = null; 

   public ErrorState(Compiler compiler) {
      this.compiler = compiler;
   }

   public void comma( ) {
      System.out.print("In error state processing "+compiler.targetState.toString( ));
      System.out.println(", tossing ','");
   }

   public void value(String str) {
      System.out.print("In error state processing "+compiler.targetState.toString( ));
      System.out.println(", tossing " + str); 
   }

   public void eol( ) { 
      System.out.print("In error state processing "+compiler.targetState.toString( ));
      System.out.println("processing eol"); 
      compiler.targetState.eol( );
   }
      
   public void eof( ) {
      System.out.print("In error state processing "+compiler.targetState.toString( ));
      System.out.println("eof found, terminating compile"); 
   }

   public String toString( ) {
      return "ErrorState";
   }
}

File: ECE39500/exams/bak_midterm-2/midterm-2/PartA/ErrorState.py
from CompilerState import CompilerState
from StateFactory import StateFactory
import sys

class ErrorState(CompilerState):
    INT_MIN = -sys.maxsize - 1

    def __init__(self, compiler):
        self.compiler = compiler
        return
    
    def comma(self):
        print(f"In error state processing {self.compiler.targetState}, tossing ','")

    def value(self, s):
        print(f"In error state processing {self.compiler.targetState}, tossing {s}")

    def eol(self):
        print(f"In error state processing {self.compiler.targetState}, processing eol")
        self.compiler.targetState.eol()

    def eof(self):
        print(f"In error state processing {self.compiler.targetState}, eof found, terminating compiler")
    
    def __str__(self):
        return "ErrorState"